src/main/java/com/darzalgames/darzalcommon/hexagon/HexagonGrid.java

```diff
@@ -0,0 +1,73 @@
+package com.darzalgames.darzalcommon.hexagon;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.darzalgames.darzalcommon.data.BiMap;
+import com.darzalgames.darzalcommon.data.Coordinate;
+
+public abstract class HexagonGrid {
```

Comment:
This class is fine as it is, but since coordinates and Hexagons are so similar, you could replace the `Bimap` with a `Set` potentially, and also, replace the sub classes with factory methods for creating different hexagon shapes. As well, you could also make HexagonGrid a generic class directly, like `Hexagon<E>`

Reply:
We can talk about it